tests/integration/test_s3.py

```diff
@@ -1625,6 +1625,33 @@ def test_presigned_url_signature_authentication(self):
         finally:
             config.S3_SKIP_SIGNATURE_VALIDATION = old_config
 
+    def add_query_param(self, request, **kwargs):
+        request.url += 'requestedBy=abcDEF123'
+
+    def test_presign_with_query_params(self):
+        bucket_name = short_uid()
+        s3_client = aws_stack.connect_to_service('s3')
+        s3_presign = boto3.client('s3',
+            endpoint_url=config.get_edge_url(),
+            aws_access_key_id='test',
+            aws_secret_access_key='test',
+            config=Config(signature_version='s3v4')
+        )
+
+        s3_client.create_bucket(Bucket=bucket_name)
+        s3_client.put_object(Body='test-value', Bucket=bucket_name, Key='test')
+        response = s3_client.head_object(Bucket=bucket_name, Key='test')
+        s3_client.meta.events.register('before-sign.s3.GetObject', self.add_query_param)
```

Comment:
nit: let's better wrap this code in a `try-finally` block (running the `unregister(..)` below in the `finally` block), to ensure that the event handler is unregistered even if this tests raises an error.

Also, let's add a short note nere that this test may have side effects (and hence may be problematic for parallelization):
```
# Note: This test may have side effects (via `s3_client.meta.events.register(..)`) and may not be suitable for parallel execution
```

Reply:
Thanks for the suggestions. I have updated the code accordingly.